pkg/cloudprovider/providers/gce/gce.go

```diff
@@ -397,31 +398,49 @@ func CreateGCECloud(config *CloudConfig) (*GCECloud, error) {
 
 	// ProjectID and.NetworkProjectID may be project number or name.
 	projID, netProjID := tryConvertToProjectNames(config.ProjectID, config.NetworkProjectID, service)
-
 	onXPN := projID != netProjID
 
 	var networkURL string
 	var subnetURL string
+	var isLegacyNetwork bool
 
-	if config.NetworkName == "" && config.NetworkURL == "" {
-		// TODO: Stop using this call and return an error.
-		// This function returns the first network in a list of networks for a project. The project
-		// should be set via configuration instead of randomly taking the first.
-		networkName, err := getNetworkNameViaAPICall(service, config.NetworkProjectID)
-		if err != nil {
-			return nil, err
-		}
-		networkURL = gceNetworkURL(config.ApiEndpoint, netProjID, networkName)
-	} else if config.NetworkURL != "" {
+	if config.NetworkURL != "" {
 		networkURL = config.NetworkURL
-	} else {
+	} else if config.NetworkName != "" {
 		networkURL = gceNetworkURL(config.ApiEndpoint, netProjID, config.NetworkName)
+	} else {
+		glog.Warningf("No network name or URL specified.")
```

Comment:
What happens in this case, should we return error? Or is the `subnetURL` below all we need?

Reply:
Depends on what the user of this package wants to do. Kubelet inits the cloud provider but only uses a static function that's not related to networking. 

I'm not convinced that we can return an error here and not have this bite us in the future.

Reply:
Kubelet's use of cloud provider seems to be unnecessary. But does it make sense to have a cloud provider be initialized but not fully functional?

Reply:
It's indeed unnecessary, but we have to live with it for now.  Since gce.conf is not passed to kubelet, the cloudprovider is already mostly defunct. `CreateGCECloud` needs to not throw an error despite the lack of functionality. Meanwhile, ideally we *do* throw an error if we get into a defunct state for service/ingress controller. I don't think there's a great answer here until we limit the consumers of cloudprovider.

I'm hesitant to return an error here because I'm worried about the case of the metadata API call not returning the network name. This would cause kubelet to not start.